lib/JMAP/Tester/Response.pm

```diff
@@ -37,11 +37,39 @@ has wrapper_properties => (
 sub items { @{ $_[0]->_items } }
 
 sub add_items {
-  $_[0]->sentence_broker->abort_callback->("can't add items to " . __PACKAGE__);
+  $_[0]->sentence_broker->abort("can't add items to " . __PACKAGE__);
 }
 
-sub sentence_broker {
-  state $BROKER = JMAP::Tester::SentenceBroker->new;
+sub default_diagnostic_dumper {
+  state $default = do {
+    require JSON::MaybeXS;
+    state $json = JSON::MaybeXS->new->utf8->convert_blessed->pretty->canonical;
+    sub { $json->encode($_[0]); }
+  };
+
+  return $default;
 }
 
+has _diagnostic_dumper => (
+  is => 'ro',
+  builder   => 'default_diagnostic_dumper',
+  init_arg  => 'diagnostic_dumper',
+);
```

Comment:
Should this be lazy? Otherwise I don't think requiring JSON::MaybeXS at run-time gives us too much benefit